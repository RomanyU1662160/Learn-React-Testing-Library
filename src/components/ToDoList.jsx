import React, { useState } from "react";
import ToDoItem from "./ToDoItem";

const ToDoList = props => {
  const { list, deleteTodo } = props;

  console.log("list in todolist ", list);
  return (
    <div>
      <h3> to do list List</h3>
      {list.map(item => (
        <ToDoItem key={item.id} item={item} deleteTodo={deleteTodo}></ToDoItem>
      ))}
    </div>
  );
};

export default ToDoList;
