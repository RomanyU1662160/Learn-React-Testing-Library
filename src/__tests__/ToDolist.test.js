import React from "react";
import { render, cleanup, waitForElement } from "@testing-library/react";
import "@testing-library/jest-dom";
import ToDoList from "../components/ToDoList";

const init = [{ id: 1, name: "Learn React.js", state: false }];

afterEach(cleanup);
it("render the h3 title ", () => {
  const { getByTestId, getByText, asFragment } = render(
    <ToDoList list={init}></ToDoList>
  );
  expect(asFragment()).toMatchSnapshot();

  const title = getByTestId("listTitle");
  expect(title).toHaveTextContent("The To Do List");

  const byText = getByText("The To Do List");
  expect(byText).toBeInTheDocument();
  expect(byText).toHaveClass("text-center");
});

it("tests the init list", () => {
  const { getByTestId } = render(<ToDoList list={init}></ToDoList>);

  expect(getByTestId("todo")).toHaveTextContent("Learn React.js");
});

// it("test the list props", async () => {
//   const { getByText } = render(<ToDoList list={init}> </ToDoList>);
//   const byText = await waitForElement(() => getByText("Learn React.js"));
//   //  expect(byText.value).toHaveTextContent("Learn React.js");
//   expect(byText.value).toBeInTheDOM();
// });
