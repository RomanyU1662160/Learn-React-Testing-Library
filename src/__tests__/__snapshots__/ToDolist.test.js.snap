// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render the h3 title  1`] = `
<DocumentFragment>
  <div>
    <h3
      class="text-info text-center"
      data-testid="listTitle"
    >
      The To Do List
    </h3>
    <div
      class="row border-bottom p-4"
    >
      <div
        class="col-md-4"
        data-testid="todo"
      >
         Name : Learn React.js 
      </div>
      <div
        class="col-md-4"
      >
        Status:  not Yet 
      </div>
      <div
        class="col-md-2"
      >
        <button
          class="btn btn-danger"
        >
          Delete
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
